app/src/main/java/app/gamenative/service/gog/GOGService.kt

```diff
@@ -0,0 +1,359 @@
+        }
+
+
+        suspend fun refreshSingleGame(gameId: String, context: Context): Result<GOGGame?> {
+            return getInstance()?.gogManager?.refreshSingleGame(gameId, context)
+                ?: Result.failure(Exception("Service not available"))
+        }
+
+        /**
+         * Delete/uninstall a GOG game
+         * Delegates to GOGManager.deleteGame
+         */
+        suspend fun deleteGame(context: Context, libraryItem: LibraryItem): Result<Unit> {
+            return getInstance()?.gogManager?.deleteGame(context, libraryItem)
+                ?: Result.failure(Exception("Service not available"))
+        }
+    }
+
+    private lateinit var notificationHelper: NotificationHelper
+
+    @Inject
+    lateinit var gogManager: GOGManager
+
+    private val scope = CoroutineScope(Dispatchers.IO + SupervisorJob())
+
+    // Track active downloads by game ID
+    private val activeDownloads = ConcurrentHashMap<String, DownloadInfo>()
+
+    // GOGManager is injected by Hilt
+    override fun onCreate() {
+        super.onCreate()
+        instance = this
+
+
+        // Initialize notification helper for foreground service
+        notificationHelper = NotificationHelper(applicationContext)
+    }
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        Timber.d("[GOGService] onStartCommand() - action: ${intent?.action}")
+
+        // Start as foreground service
+        val notification = notificationHelper.createForegroundNotification("GOG Service running...")
+        startForeground(2, notification) // Use different ID than SteamService (which uses 1)
+
+        // Start background library sync if service is starting or sync action requested
+        if (intent?.action == ACTION_SYNC_LIBRARY || backgroundSyncJob == null || !backgroundSyncJob!!.isActive) {
+            Timber.i("[GOGService] Triggering background library sync")
+            backgroundSyncJob?.cancel() // Cancel any existing job
+            backgroundSyncJob = scope.launch {
+                try {
+                    setSyncInProgress(true)
+                    Timber.d("[GOGService]: Starting background library sync")
+
+                    val syncResult = gogManager.startBackgroundSync(applicationContext)
+                    if (syncResult.isFailure) {
+                        Timber.w("[GOGService]: Failed to start background sync: ${syncResult.exceptionOrNull()?.message}")
+                    } else {
+                        Timber.i("[GOGService]: Background library sync started successfully")
+                    }
+                } catch (e: Exception) {
+                    Timber.e(e, "[GOGService]: Exception starting background sync")
+                } finally {
+                    setSyncInProgress(false)
+                }
+            }
+        } else {
+            Timber.d("[GOGService] Background sync already in progress, skipping")
+        }
+
+        return START_STICKY
```

Comment:
Also, we need to close the GOG Service when the app is closed. It runs indefinitely rn:
<img width="614" height="347" alt="image" src="https://github.com/user-attachments/assets/337e2c14-7ad1-423d-b2ab-740ce5c44695" />


We do this in MainActivity for Steam, can use the same logic. This will kill the battery otherwise.